utils/ansible.py

```diff
@@ -0,0 +1,269 @@
+            'provider_api_auth_token': get_crud('openshift-3').credentials['token'].token,
+            'hawkular_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', [])
+        }
+    elif script_type == 'users':
+        return {
+            'fullname': 'MIQUser',
+            'name': 'MIQU',
+            'password': 'smartvm',
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+    elif script_type == 'custom_attributes':
+        return {
+            'entity_type': 'provider',
+            'entity_name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+
+
+# TODO Avoid reading files every time
+def read_yml(script, value):
+    with open(yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def get_yml_value(script, value):
+    with open(basic_yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def setup_basic_script(provider, script_type):
+    copyfile(basic_yml_path + script_type + "_basic_script.yml",
+             basic_yml_path + script_type + "_basic_script_bk.yml")
+    with open(basic_yml_path + script_type + "_basic_script.yml", 'rw') as f:
+        doc = yaml.load(f)
+        values_dict = get_values_from_conf(provider, script_type)
+    for key in values_dict:
+        if script_type == 'providers':
+            doc[0]['tasks'][0]['manageiq_provider'][key] = values_dict[key]
+        elif script_type == 'users':
+            doc[0]['tasks'][0]['manageiq_user'][key] = values_dict[key]
+        elif script_type == 'custom_attributes':
+            doc[0]['tasks'][0]['manageiq_custom_attributes'][key] = values_dict[key]
+        with open(basic_yml_path + script_type + "_basic_script.yml", 'w') as f:
+            f.write(yaml.dump(doc))
+
+
+def open_yml(script, script_type):
+    copyfile(basic_yml_path + "/" + script_type + "_basic_script.yml",
+             basic_yml_path + script + ".yml")
+    with open(basic_yml_path + script + ".yml", 'rw') as f:
+        return yaml.load(f)
+
+
+def write_yml(script, doc):
+    with open(basic_yml_path + script + ".yml", 'w') as f:
+        f.write(yaml.dump(doc))
+
+
+def setup_ansible_script(provider, script, script_type=0, values_to_update=0, user_name=0):
+    # This function prepares the ansible scripts to work with the correct
+    # appliance configs that will be received from Jenkins
+    setup_basic_script(provider, script_type)
+    if script == 'add_provider':
```

Comment:
**OPTIONAL:** Perhaps a better way to have this organized is to class-ify it and make each of these a method so that then a) people can use ipython to see which methods are available and b) you are not having to parse a massive conditional with your eyes.